internal/definitions/definition_handler_identity_claim.go

```diff
@@ -0,0 +1,218 @@
+	valid := dh.getSystemBroadcastPayload(ctx, msg, data, &claim)
+	if !valid {
+		return ActionReject, nil
+	}
+
+	return dh.handleIdentityClaim(ctx, state, msg, &claim, verificationID)
+
+}
+
+func (dh *definitionHandlers) verifyClaimSignature(ctx context.Context, msg *fftypes.Message, identity *fftypes.Identity, parent *fftypes.Identity) (valid bool) {
+
+	author := msg.Header.Author
+	if author == "" {
+		return false
+	}
+
+	var expectedSigner *fftypes.Identity
+	switch {
+	case identity.Type == fftypes.IdentityTypeNode:
+		// In the special case of a node, the parent signs it directly
+		expectedSigner = parent
+	default:
+		expectedSigner = identity
+	}
+
+	valid = author == expectedSigner.DID ||
+		(expectedSigner.Type == fftypes.IdentityTypeOrg || author == fmt.Sprintf("%s%s", fftypes.FireFlyOrgDIDPrefix, expectedSigner.ID))
+	if !valid {
+		log.L(ctx).Warnf("Unable to process identity claim %s - signature mismatch type=%s author=%s expected=%s", msg.Header.ID, identity.Type, author, expectedSigner.DID)
+	}
+	return valid
+}
+
+func (dh *definitionHandlers) getClaimVerifier(msg *fftypes.Message, identity *fftypes.Identity) *fftypes.Verifier {
+	verifier := &fftypes.Verifier{
+		ID:        fftypes.NewUUID(),
+		Identity:  identity.ID,
+		Namespace: identity.Namespace,
+	}
+	switch identity.Type {
+	case fftypes.IdentityTypeNode:
+		verifier.VerifierRef.Type = fftypes.VerifierTypeFFDXPeerID
+		verifier.VerifierRef.Value = identity.Profile.GetString("id")
+	default:
+		verifier.VerifierRef.Type = dh.blockchain.VerifierType()
+		verifier.VerifierRef.Value = msg.Header.Key
+	}
+	return verifier
+}
+
+func (dh *definitionHandlers) confirmVerificationForClaim(ctx context.Context, state DefinitionBatchState, msg *fftypes.Message, identity, parent *fftypes.Identity) (*fftypes.UUID, error) {
+	// Query for messages on the topic for this DID, signed by the right identity
+	idTopic := identity.Topic()
+	fb := database.MessageQueryFactory.NewFilter(ctx)
+	filter := fb.And(
+		fb.Eq("topics", idTopic),
+		fb.Eq("author", parent.DID),
+		fb.Eq("type", fftypes.MessageTypeDefinition),
+		fb.Eq("state", fftypes.MessageStateConfirmed),
+		fb.Eq("tag", fftypes.SystemTagIdentityVerification),
+	)
+	candidates, _, err := dh.database.GetMessages(ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+	// We also need to check pending messages in the current pin batch
+	for _, pending := range state.GetPendingConfirm() {
+		if pending.Header.Topics.String() == idTopic &&
+			pending.Header.Author == parent.DID &&
+			pending.Header.Type == fftypes.MessageTypeDefinition &&
+			pending.Header.Tag == fftypes.SystemTagIdentityVerification {
+			candidates = append(candidates, pending)
+		}
+	}
+	for _, candidate := range candidates {
+		data, foundAll, err := dh.data.GetMessageData(ctx, candidate, true)
+		if err != nil {
+			return nil, err
+		}
+		identityMatches := false
+		var verificationID *fftypes.UUID
+		var verificationHash *fftypes.Bytes32
```

Comment:
minor note, but it's not necessary to pre-declare these 3 vars, is it?

Reply:
It allows us to have consistent logging outside of the `foundAll` branch